src/android/io/branch/BranchSDK.java

```diff
@@ -226,19 +219,11 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
      */
     private void initSession(CallbackContext callbackContext)
     {
-        Log.d(LCAT, "start initSession()");
-
         this.activity = this.cordova.getActivity();
 
         this.instance = Branch.getAutoInstance(this.activity.getApplicationContext());
 
-        if (callbackContext == null) {
-            this.instance.initSession(new SessionListener(), activity.getIntent().getData(), activity);            
-        } else {
-            this.instance.initSession(new SessionListener(callbackContext), activity.getIntent().getData(), activity);
-        }
-
-
+        this.instance.initSession(new SessionListener(callbackContext), activity.getIntent().getData(), activity);
```

Comment:
The if/else statement should not be removed.
This was a fixed for `initSession` for `onNewIntent`condition, wherein it doesn't have a callback instance being passed.

See this [PR](https://github.com/BranchMetrics/Cordova-Ionic-PhoneGap-Deferred-Deep-Linking-SDK/pull/103/files)